BetterShardsBukkit/src/main/java/vg/civcraft/mc/bettershards/database/DatabaseManager.java

```diff
@@ -1019,6 +1019,64 @@ public void removeBed(UUID uuid) {
 			logger.log(Level.SEVERE, "Failed to removeBed, exception:", e);
 		}
 	}
+	
+	/**
+	 * This method is used to register with BetterShards what portal_id should
+	 * be associated with each portal so when saving occurs each portal specific id
+	 * is respected.
+	 * This method should be called for each portal type that exists.
+	 * So what should happen is in each plugin you give your custom portal it's own
+	 * specific id that is internally recognized and tracked. From there you will pass
+	 * that id and your plugin name to this method and it will create an id that can be 
+	 * used to pass in your constructor for your portal objects.
+	 * After calling this method refer to the 
+	 * {@link #getPortalID(String, int)} method for getting what id to pass to 
+	 * your custom portal constructors.
+	 * This method can be called at every start and if an id is already present it will
+	 * fail silently.
+	 * @param id The id that will be used internally in your plugin to reference each 
+	 * portal type.
+	 * @param plugin Your plugin's name.
+	 */
+	public void addPortalType(int id, String plugin) {
```

Comment:
No they won't because when they get the portal id it will be specific in that specific server set up.
So better shards can create 3 portals and they will be given id 0,1,2.
Then some other plugin can create a portal and it will be given the id of 3.